unikraft/core/core.go

```diff
@@ -45,6 +45,17 @@ type UnikraftConfig struct {
 	kconfig kconfig.KeyValueMap
 }
 
+// UnikraftConfigOption is a constructor that configures a core configuration.
+func NewUnikraftConfig(opts ...UnikraftConfigOption) *UnikraftConfig {
+	uc := &UnikraftConfig{}
+
+	for _, opt := range opts {
+		opt(uc)
+	}
+
+	return uc
+}
```

Comment:
```go
uc := UnikraftConfig{}

for _, opt := range opts {
	if err := opt(&uc); err != nil {
		return nil, err
	}

	return &uc, err
}
```

Reply:
The return should include an `error` and each option should also return an error -- even if no current option has error checking, we can add error checking in the future